hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/GeneralStage.java

```diff
@@ -123,7 +145,16 @@
      * The function can return a null future or the future can return a null
      * result: in both cases it will act just like a filter.
      * <p>
-     * The latency of the async call will add to the latency of items.
+     * The latency of the async call will add to the total latency of the
+     * output.
+     * <p>
+     * Sample usage:
+     * <pre>{@code
+     * stage.mapUsingContextAsync(
+     *     ContextFactory.withCreateFn(jet -> new MyContext(jet)),
+     *     (myContext, item) -> myContext.enrichAsync(item)
```

Comment:
Function returns completable future. Make this more explicit as this is the tricky part. Such as in https://github.com/hazelcast/hazelcast-jet-code-samples/blob/master/enrichment/src/main/java/Enrichment.java#L246